ui/component/publishProtectedContent/view.jsx

```diff
@@ -0,0 +1,171 @@
+import Button from 'component/button';
+import * as PAGES from 'constants/pages';
+
+type Props = {
+  description: ?string,
+  disabled: boolean,
+  updatePublishForm: ({}) => void,
+  getMembershipTiersForContentClaimId: (type: string) => void,
+  claim: Claim,
+  protectedMembershipIds: Array<number>,
+  activeChannel: ChannelClaim,
+  incognito: boolean,
+  getExistingTiers: ({ channel_name: string, channel_id: string }) => Promise<CreatorMemberships>,
+  myMembershipTiers: CreatorMemberships,
+  myMembershipTiersWithExclusiveContentPerk: CreatorMemberships,
+  myMembershipTiersWithExclusiveLivestreamPerk: CreatorMemberships,
+  location: string,
+};
+
+function PublishProtectedContent(props: Props) {
+  const {
+    activeChannel,
+    incognito,
+    updatePublishForm,
+    getMembershipTiersForContentClaimId,
+    claim,
+    protectedMembershipIds,
+    getExistingTiers,
+    myMembershipTiers,
+    myMembershipTiersWithExclusiveContentPerk,
+    myMembershipTiersWithExclusiveLivestreamPerk,
+    location,
+  } = props;
+
+  const [isRestrictingContent, setIsRestrictingContent] = React.useState(false);
+
+  const claimId = claim?.claim_id;
+
+  let membershipsToUse = myMembershipTiersWithExclusiveContentPerk;
+  if (location === 'livestream') membershipsToUse = myMembershipTiersWithExclusiveLivestreamPerk;
+
+  // run the redux action
+  React.useEffect(() => {
+    if (claimId) {
+      getMembershipTiersForContentClaimId(claimId);
+    }
+  }, [claimId]);
+
+  // if there are already restricted memberships for this content, setup state
+  React.useEffect(() => {
+    if (activeChannel && protectedMembershipIds && protectedMembershipIds.length) {
+      setIsRestrictingContent(true);
+      const restrictionCheckbox = document.getElementById('toggleRestrictedContent');
+      // $FlowFixMe
+      if (restrictionCheckbox) restrictionCheckbox.checked = true;
+    }
+  }, [protectedMembershipIds, activeChannel, claim]);
+
+  function handleRestrictedMembershipChange(event) {
+    let matchedMemberships;
+    const restrictCheckboxes = document.querySelectorAll('*[id^="restrictToMembership"]');
+
+    // $FlowFixMe
+    for (const checkbox of restrictCheckboxes) {
+      // $FlowFixMe
+      if (checkbox.checked) {
+        matchedMemberships = new Set(matchedMemberships);
+        matchedMemberships.add(Number(checkbox.id.split(':')[1]));
+      }
+    }
+
+    const commaSeparatedValueString = matchedMemberships && Array.from(matchedMemberships).join(',');
+
+    updatePublishForm({
+      restrictedToMemberships: commaSeparatedValueString,
+      channelClaimId: activeChannel.claim_id,
+    });
+  }
+
+  function handleChangeRestriction() {
+    // update data to check against during publish
+    // backend checks against an empty string so need to use that instead of undefined
+    updatePublishForm({ restrictedToMemberships: isRestrictingContent ? '' : null });
+
+    setIsRestrictingContent(!isRestrictingContent);
+  }
```

Comment:
:small_red_triangle: `restrictedToMemberships` is set here but not treated properly in the reducer.

- (Postpone-able) Update `PublishState`.
- (Must) Either specify a default value for the slice, or clear it in `CLEAR_PUBLISH`.  Otherwise, the previous value gets carried over if the next one happens to be Anonymous.

Do update for other new variables too (if any)